components/BottomAppBar/src/private/MDCBottomAppBarView.m

```diff
@@ -0,0 +1,250 @@
+  self.floatingButton.center =
+      [self getFloatingButtonCenterPositionForWidth:CGRectGetWidth(self.bounds)];
+  [self renderPathBasedOnFloatingButtonVisibility:NO];
+}
+
+- (void)renderPathBasedOnFloatingButtonVisibility:(BOOL)animated {
+  if (self.floatingButtonVisible) {
+    [self cutBottomAppBarView:animated];
+  } else {
+    [self healBottomAppBarView:animated];
+  }
+}
+
+- (CGPoint)getFloatingButtonCenterPositionForWidth:(CGFloat)width {
+
+  CGPoint floatingButtonPoint = CGPointZero;
+  CGFloat halfDefaultDimension = CGRectGetWidth(self.floatingButton.bounds) / 2;
+  CGFloat midX = width / 2;
+
+  switch (self.floatingButtonPosition) {
+    case MDCBottomAppBarFloatingButtonPositionLeading: {
+      if (self.layoutDirection == UIUserInterfaceLayoutDirectionLeftToRight) {
+        floatingButtonPoint = CGPointMake(kMDCBottomAppBarFloatingButtonPositionX,
+                                          halfDefaultDimension);
+      } else {
+        floatingButtonPoint = CGPointMake(width - kMDCBottomAppBarFloatingButtonPositionX,
+                                          halfDefaultDimension);
+      }
+      break;
+    }
+    case MDCBottomAppBarFloatingButtonPositionCenter: {
+      floatingButtonPoint = CGPointMake(midX, halfDefaultDimension);
+      break;
+    }
+    case MDCBottomAppBarFloatingButtonPositionTrailing: {
+      if (self.layoutDirection == UIUserInterfaceLayoutDirectionLeftToRight) {
+        floatingButtonPoint = CGPointMake(width - kMDCBottomAppBarFloatingButtonPositionX,
+                                          halfDefaultDimension);
+      } else {
+        floatingButtonPoint = CGPointMake(kMDCBottomAppBarFloatingButtonPositionX,
+                                          halfDefaultDimension);
+      }
+      break;
+    }
+    default:
+      break;
+  }
+  return floatingButtonPoint;
+}
+
+- (void)cutBottomAppBarView:(BOOL)animated {
+  CGPathRef cutPath =
+      [self.bottomBarLayer pathWithCutFromRect:self.bounds
+                        floatingButtonPosition:self.floatingButtonPosition
+                               layoutDirection:self.layoutDirection];
+  if (animated) {
+    CABasicAnimation *pathAnimation =
+        [CABasicAnimation animationWithKeyPath:kMDCBottomAppBarViewPathString];
+    pathAnimation.duration = kMDCFloatingButtonExitDuration;
+    pathAnimation.fromValue = (id)self.bottomBarLayer.presentationLayer.path;
+    pathAnimation.toValue = (__bridge id _Nullable)(cutPath);
+    pathAnimation.fillMode = kCAFillModeForwards;
+    pathAnimation.removedOnCompletion = NO;
+    pathAnimation.delegate = self;
+    [pathAnimation setValue:kMDCBottomAppBarViewPathString
+                     forKey:kMDCBottomAppBarViewAnimKeyString];
+    [self.bottomBarLayer addAnimation:pathAnimation forKey:kMDCBottomAppBarViewPathString];
+  } else {
+    self.bottomBarLayer.path = cutPath;
+  }
+}
+
+- (void)healBottomAppBarView:(BOOL)animated  {
```

Comment:
Per UIKIt, add Animated: to method name.

Reply:
Done.